### PR TITLE
Propose Individual profile

### PR DESCRIPTION
**Pull request by goneall**
This is one possible solution to issue #306 

The intent is require `creationInfo` on the `Bundle` class and create a profile which requires `creationInfo` on all `Element`s.

This would allow the typical use case of providing a `Bundle` of elements to a user to only be required to specify the `creationInfo` once.  It would also provide producers the option of specifying a `individual` profile which would require `creationInfo` for all `Elements`.  This would be used when a producer provides `Elements` which are not contained in a `Bundle`.

Note that `SpdxDocument` is a subclass of `Bundle` so this approach would be compatible with SPDX 2.X.

### CODE REVIEW

**Comment by goneall** [comment]
@sbarnum @tsteenbe @nishakm @kestewart Let me know what you think.

File: model/Individual/Classes/Element.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+SPDX-License-Identifier: Community-Spec-1.0
```
Comment (zvr) [inline model/Individual/Classes/Element.md:1]: The convention is that you don't create a new class file (as there is not really a new class).

You put the `External properties restrictions` info on the top-level (namespace) file.

File: model/Individual/Individual.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+## Description
+
+Although most use cases involve sharing `Elements` inside a `Collection`, there are cases where a stand-alone SPDX element may be made available.
+In those cases, the `Element` `creationInfo` property would be required to identify the `Entity` that created the element.
+
+## Metadata
+
+- id: https://rdf.spdx.org/v3/Individual
+- name: Individual
```
Comment (zvr) [inline model/Individual/Individual.md:17]: you should add the `External properties restrictions` info here.

**Review by zvr — CHANGES_REQUESTED** [review]

File: model/Core/Classes/Bundle.md [hunk 3]
```diff
@@ -22,3 +22,7 @@ A bundle is a collection of Elements that have a shared context.
   - type: xsd:string
   - maxCount: 1
 
+## External properties restrictions
```
[the remark this answers is not in the document]
Comment (goneall) [reply model/Core/Classes/Bundle.md:25]: I did change my mind after reading the descriptions. I thought that `Bundle` being a subclass of `ElementCollection` and a superclass of `SpdxDocument` being defined as having a "shared context" was a more appropriate to place a restriction.  The reasoning being that if a `Bundle` does not have a shared context, it probably shouldn't be required to share the `creationInfo`.

**Comment by goneall** [comment]
> Now that I think about it, since one can share many elements, not only a single one, maybe call this unbundled profile?

I like it better than `Individual`.  I'll wait until after next week's tech call to rename in case we come up with something even better.

File: model/Individual/Individual.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+## External properties restrictions
+
+- /Core/creationInfo
+  - type: /Core/Element/CreationInfo
+  - minCount: 1
+  - maxCount: 1
```
Comment (zvr) [inline model/Individual/Individual.md:24]: This is not the right way. Please use (to mark class, property, and only the values changed):

```
- /Core/Element/creationInfo
  - minCount: 1
```

**Review by zvr — CHANGES_REQUESTED** [review]

File: model/Individual/Individual.md (hunk 4 above)
Comment (goneall) [reply model/Individual/Individual.md:24]: So - is the type /Core/CreationInfo?

File: model/Individual/Individual.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# Individual
+
+## Summary
+
+A profile to be used when sharing individual SPDX `Element`.
+
+## Description
+
+Although most use cases involve sharing `Elements` inside a `Collection`, there are cases where a stand-alone SPDX element may be made available.
+In those cases, the `Element` `creationInfo` property would be required to identify the `Entity` that created the element.
```
Comment (armintaenzertng) [inline model/Individual/Individual.md:12]: `Entity` will be called `Agent` soon.

File: model/Individual/Individual.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# Individual
+
+## Summary
+
+A profile to be used when sharing individual SPDX `Element`.
```
Comment (armintaenzertng) [inline model/Individual/Individual.md:7]:
```suggestion
A profile to be used when sharing individual SPDX Elements.
```
I'm not sure we use backticks anywhere else in the markdown files but so far, it does not seem the standard. I think there was a problem with them not working well with the automated documentation, @maxhbr?

File: model/Individual/Individual.md (hunk 4 above)
Comment (armintaenzertng) [reply model/Individual/Individual.md:24]: As you don't change the type of the external property, I don't think you have to specify it here (the same goes for maxCount).

Comment (zvr) [reply model/Individual/Individual.md:24]: Right, for external property restrictions, we use _/namespace/class/property_ to point to something and then we list the changed values.

See, for example, how `File` specifies that `name` (a property of `Element`) is mandatory -- this is exactly the same.

File: model/Individual/Individual.md (hunk 5 above)
Comment (goneall) [reply model/Individual/Individual.md:12]: Agree - I left it as the current definition to match the repo

File: model/Individual/Individual.md (hunk 6 above)
Comment (goneall) [reply model/Individual/Individual.md:7]: I'll remove the back ticks

File: model/Individual/Individual.md (hunk 5 above)
Comment (goneall) [reply model/Individual/Individual.md]: It has now been updated to Agent

**Comment by goneall** [comment]
Based on discussions on compaction, we decided that having a reference to a common creation info was the preferred path.  Closing this issue.

**Pull request closed without merging**